Error when compiling with hashable-1.3.4.1

Database/Beam/Haskell/Syntax.hs:1041:10: error:
    Duplicate instance declarations:
      instance Hashable a => Hashable (S.Set a)
        -- Defined at Database/Beam/Haskell/Syntax.hs:1041:10
      instance Hashable v => Hashable (S.Set v)
        -- Defined in ‘hashable-1.3.4.1:Data.Hashable.Class’
     |
1041 | instance Hashable a => Hashable (S.Set a) where
     |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

--- a/beam-migrate/Database/Beam/Haskell/Syntax.hs
+++ b/beam-migrate/Database/Beam/Haskell/Syntax.hs
@@ -1038,5 +1038,7 @@ instance Hashable (Hs.CName ())
 instance Hashable (Hs.DerivStrategy ())
 instance Hashable (Hs.MaybePromotedName ())
 #endif
+#if !MIN_VERSION_hashable(1, 3, 4)
 instance Hashable a => Hashable (S.Set a) where
   hashWithSalt s a = hashWithSalt s (S.toList a)
+#endif

